### PR TITLE
AMPUI-80 call activity endpoint. load fields definition, load possibl…

### PR DESCRIPTION
**Pull request by jdeanquin-dg**
…e values, load FM tree, refactor common possiblevalues to ampui

### CODE REVIEW

File: amp/TEMPLATE/reamp/modules/activity/preview/containers/App.jsx [hunk 1]
```diff
@@ -54,31 +51,34 @@ class App extends Component {
...
         }
     }
 
     render() {
-        return (
-            <ActivityPreviewUI activity={activity} activityContext={activityContext} />
-        );
+        console.log(this.props);
+        if (this.props.activityReducer.isActivityLoading) {
+            return (<div> LOADING ACTIVITY </div>)
```
Comment (ginchauspe) [inline amp/TEMPLATE/reamp/modules/activity/preview/containers/App.jsx]: In a future phase we will add translate here.

File: amp/TEMPLATE/reamp/modules/activity/preview/api/ActivityApi.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import {fetchJson, postJson} from 'amp/tools/index';
+import {ACTIVITY_API, FIELDS_DEFINITION_API, POSSIBLE_VALUES_API, FM_API} from '../common/Constants.jsx'
+
+
+export default class ActivityApi {
+    static getActivity(activityId) {
+        return new Promise((resolve, reject) => {
+            fetchJson(ACTIVITY_API + activityId).then((activity) => {
```
Comment (ginchauspe) [inline amp/TEMPLATE/reamp/modules/activity/preview/api/ActivityApi.jsx]: my ES6 is a bit rusty, could you confirm if in these cases is not necessary to put a return before beforeJson?

File: amp/TEMPLATE/reamp/modules/activity/preview/containers/App.jsx (hunk 1 above)
Comment (jdeanquin-dg) [reply amp/TEMPLATE/reamp/modules/activity/preview/containers/App.jsx]: yes once we implement the translate module we will. Good catch

File: amp/TEMPLATE/reamp/modules/activity/preview/api/ActivityApi.jsx (hunk 2 above)
Comment (jdeanquin-dg) [reply amp/TEMPLATE/reamp/modules/activity/preview/api/ActivityApi.jsx]: I think we do, let me check. Also we could enable eslint but only for the preview module

**Pull request merged**